test/gtest/gtest_common.hpp

```diff
@@ -171,6 +171,15 @@ void invoke_with_params(Check&& check)
     }
 }
 
+/// repalcement of gtest's FAIL() because FAIL() returns void and messes up the
+/// return type deduction
+#define MIOPEN_LOG_FATAL(...)                                 \
+    do                                                        \
+    {                                                         \
+        MIOPEN_LOG(miopen::LoggingLevel::Fatal, __VA_ARGS__); \
+        std::abort();                                         \
+    } while(false);
+
```

Comment:
```suggestion
#define MIOPEN_FRIENDLY_FAIL(MSG)                             \
    do                                                        \
    {                                                         \
        [&]() { FAIL() << MSG;}();                            \
    } while(false);
```
I don't think that `std::abort` is a good solution. Gtest's `FAIL()` does not abort and switches to the next test, but in this case it messes whole gtest functionality. I would say the same about C `assert`.

BTW, what is the difference between `assert` and `MIOPEN_LOG_FATAL`? Shall `assert` be replaced?

Reply:
Good point. I did not know how `FAIL()` was different from `abort()`. We can replace asserts with `MIOPEN_FRIENDLY_FAIL()` I believe.